nautobot/extras/jobs.py

```diff
@@ -1239,3 +1242,47 @@ def scheduled_job_handler(*args, **kwargs):
 
     job_content_type = get_job_content_type()
     JobResult.enqueue_job(run_job, name, job_content_type, user, schedule=schedule, **kwargs)
+
+
+def enqueue_job_hooks(object_change):
+    """
+    Find job hook(s) assigned to this changed object type + action and enqueue them
+    to be processed
+    """
+    from nautobot.extras.models import JobResult  # avoid circular import
+
+    # Job hooks cannot trigger other job hooks
+    if object_change.change_context == ObjectChangeEventContextChoices.CONTEXT_JOB_HOOK:
+        return
+
+    # Determine whether this type of object supports webhooks
+    app_label = object_change.changed_object._meta.app_label
+    model_name = object_change.changed_object._meta.model_name
+    if model_name not in registry["model_features"]["webhooks"].get(app_label, []):
+        return
+
+    # Retrieve any applicable job hooks
+    content_type = ContentType.objects.get_for_model(object_change.changed_object)
+    action_flag = {
+        ObjectChangeActionChoices.ACTION_CREATE: "type_create",
+        ObjectChangeActionChoices.ACTION_UPDATE: "type_update",
+        ObjectChangeActionChoices.ACTION_DELETE: "type_delete",
+    }[object_change.action]
```

Comment:
Not that it's a huge deal, but this being defined on every call makes me feel like this mapping might be better as a module-level constant.